porcupine/plugins/run/no_terminal.py

```diff
@@ -258,6 +278,19 @@ def stop_executor(self, junk_event: object = None, *, quitting: bool = False) ->
         if self.executor is not None:
             self.executor.stop(quitting=quitting)
 
+    def pause_resume_executor(self, junk: object = None) -> None:
+        if sys.platform == "win32" or self.executor is None:
+            return
+
+        if self.executor.paused:
+            self.executor.send_signal(signal.SIGCONT)
```

Comment:
I believe it is enough to send `SIGSTOP` and `SIGCONT` to the parent process. For example, let's create a `bash` process with a child process `sleep 10`, and stop only the parent:

```
>>> has_sleeping_child = subprocess.Popen(['bash', '-c', 'sleep 10'])
>>> has_sleeping_child.send_signal(signal.SIGSTOP)
```

If I now wait for more than 10 seconds, I can still see the child process in the process list, meaning it stopped too:

```
>>> subprocess.call('ps aux | grep sleep', shell=True)
akuli     7675  0.0  0.0   5308   580 pts/0    T+   23:29   0:00 sleep 10
akuli     7699  0.0  0.0   6200   720 pts/0    S+   23:30   0:00 grep sleep
0
```

Reply:
I tested it with Tkinter code, and it only paused the parent process (`sh`), but not Tkinter (`python3`)

Reply:
This works (freezes the GUI):

```
akuli@akuli-desktop:~$ bash -c 'python3 -c "import tkinter; tkinter.Tk().mainloop()"' &
[1] 9741
akuli@akuli-desktop:~$ kill -STOP 9741
```

This doesn't work:

```
akuli@akuli-desktop:~$ sh -c 'python3 -c "import tkinter; tkinter.Tk().mainloop()"' &
[1] 9753
akuli@akuli-desktop:~$ kill -STOP 9753
```

Apparently bash does some magic that makes `SIGSTOP` apply to its child processes too, and `sh` doesn't do it. A few options:
- Figure out what exactly the magic is (with e.g. `strace`, looking around in `/proc`, or googling)
- Change the shell we use from sh to bash
- Keep the complicated child process logic (although I'd rather not mess with the user's child processes if we can avoid that)

Reply:
In either case, I think we can improve that in a separate PR :)

Reply:
I figured out what the "magic" is by looking at the output of `pstree -a`. Here I started `bash -c 'python3 -c "..."' &` from `sh`:

```
`-sh
    |-less
    |-pstree -a
    `-python3 -c import tkinter; tkinter.Tk().mainloop()
        `-{python3}
```

There's no `bash` process! When I ran `bash`, it replaced itself with (using `exec`) Python 3, apparently because that's the only thing it was supposed to run. I get a different result if I tell bash to also `sleep 1`:

```
`-sh
    |-bash -c python3 -c "import tkinter; tkinter.Tk().mainloop()"; sleep 1
    |   `-python3 -c import tkinter; tkinter.Tk().mainloop()
    |       `-{python3}
    |-less
    `-pstree -a
```

And if I now kill with `SIGSTOP`, the GUI doesn't freeze. So the "magic" that bash did was replacing itself with the Python or `sleep` process, so I was really stopping the Python or `sleep` directly.

Ideally there would be a way to disable this behavior so that no matter what commands the user runs, our only-one-level-deep killing logic will not recurse into child processes created by the user's programs.